### ISSUE TITLE
[TRACKING] discussion & planning for future of `kubeflow/kubeflow` repo

### ISSUE THREAD

**Issue by thesuperzapper** [kubeflow/kubeflow#7549]
As part of https://github.com/kubeflow/internal-acls/pull/618 (giving @kimwnasptd and @thesuperzapper write access to the `kubeflow/kubeflow` repo), we were asked by the @kubeflow/kubeflow-steering-committee to make a plan for the future of the `kubeflow/kubeflow` repo.

## Background

All components that currently live in `kubeflow/kubeflow` (under the [`./components/`](https://github.com/kubeflow/kubeflow/tree/master/components) folder) are owned and maintained by @kubeflow/wg-notebooks-leads.

People have identified a few issues with this:

1. I can make development harder:
    - We want to release our components more frequently than Kubeflow itself _(which may create confusion for users, if they see a `2.X.X` release on the `kubeflow/kubeflow` repo)_
    - Until recently, we didn't even have write access to the repo _(e.g. could not create new branches, tags, or cherry-pick commits)_
2. New users are sometimes confused about where to find source code

## Options

There is not a clear "best option" for the future of the `kubeflow/kubeflow` repo, but here are the 3 ones I can see.
 
__NOTE:__ any plan that involves moving code will break ALL pending PRs on that code, in addition to many historical links (unless we include README files to say the code is moved).

### Option 1: do nothing

We could just leave everything as is.

The existing code has lived for so long in its current location, and we can address most concerns with better documentation.

### Option 2: move non-core components

We could leave the "core components" in the `kubeflow/kubeflow` repo, and move everything else to a new `kubeflow/workspaces` repo.

We don't have to call it `workspaces`, but that name seems appropriate.

That would leave us with the following:

- `kubeflow/kubeflow`:
     - [`access-management`](https://github.com/kubeflow/kubeflow/tree/master/components/access-management) (KFAM, Auth)
     - [`admission-webhook`](https://github.com/kubeflow/kubeflow/tree/master/components/admission-webhook) (PodDefaults)
     - [`centraldashboard`](https://github.com/kubeflow/kubeflow/tree/master/components/centraldashboard)
     - [`profile-controller`](https://github.com/kubeflow/kubeflow/tree/master/components/profile-controller)
- `kubeflow/workspaces`:
     - [`crud-web-apps`](https://github.com/kubeflow/kubeflow/tree/master/components/crud-web-apps) (UIs for: Volumes, Notebooks, Tensorboards)
     - [`example-notebook-servers`](https://github.com/kubeflow/kubeflow/tree/master/components/example-notebook-servers) (pre-built Docker images for Notebooks)
         - (this would probably live in a separate branch of `kubeflow/workspaces` so it could be versioned separately)
     - [`notebook-controller`](https://github.com/kubeflow/kubeflow/tree/master/components/notebook-controller)
     - [`pvcviewer-controller`](https://github.com/kubeflow/kubeflow/tree/master/components/pvcviewer-controller)
     - [`tensorboard-controller`](https://github.com/kubeflow/kubeflow/tree/master/components/tensorboard-controller)

I feel like this makes sense, because the core components will still logically be versioned with the overall releases of Kubeflow.

NOTE: all the `kubeflow/workspaces` components will still depend on the core components (that is, they can't be deployed separately, and it is unlikely they could be made to).

### Option 3: move everything

If we are ideologically opposed to having anything in `kubeflow/kubeflow` we could also move the core components to their own repo.

However, I think there is little benefit to this extra effort, and it will also make the `kubeflow/kubeflow` repo have no regular commits (which will hurt SEO and GitHub stars).

**Comment by andreyvelich** [comment]
Thank you for opening this @thesuperzapper!
Please can you also add the suggestion that @kimwnasptd proposed here: https://github.com/kubeflow/internal-acls/pull/618#issuecomment-2036432934

**Comment by juliusvonkohout** [comment]
I would like to add @andreyvelich option mentioned here https://github.com/kubeflow/internal-acls/pull/618#issuecomment-2040030707

My personal opinion is 

The more repositories we have, the more cumbersome development becomes. There is also the requirement from the KSC to split kubeflow/kubeflow. kubeflow/control-plane + kubeflow/workspaces and kubeflow/manifests adds too much  development and synchronization overhead in my opinion. Moving multi-user/multi-tenancy stuff into Kubeflow/manifests where already most of the multi tenancy stuff lives makes more sense to me. Splitting what we have to release together anyway over multiple repositories does not make that much sense to me. Kfam, profile controller, etc. are tightly coupled with kubeflow/manifests. I am also fine with renaming kubeflow/manifests to kubeflow/control-plane or so. But we need as few repositories as possible and a common place for multi-user stuff. For example notebooks, PVC-viewer and maybe some other things could stay in kubeflow/workspaces.

**Comment by kimwnasptd** [comment]
The above is to immediately unblock the effort of cleaning up `kubeflow/kubeflow` for now. Then the next goal should be to help on the scope of what the Notebooks WG owns, and what is included in the `kubeflow/workspaces` repo.

Specifically I would suggest we think about and have answers with the @kubeflow/kubeflow-steering-committee on the following:
1. What do we do with components that aim to make it more streamlined to interact with K8s? (poddefaults, volumes web app for managing pvcs, pvcviewer controller)
2. TensorBoard is an ML tool, but doesn't really fit to live under `kubeflow/workspaces`. Let's either
	1. Deprecate the controller and web app if it's not used a lot (should we have a survey? cc @StefanoFioravanzo)
	2. Have a new repo, in the future, for this and potentially other Data Visualisation tools

IMO with answering the above will also help the `kubeflow/workspaces` repo to be more clean by focusing it more on notebooks/workspaces and not on other K8s functionalities or data visualisation tools.

**Comment by andreyvelich** [comment]
cc @kubeflow/wg-pipeline-leads @kubeflow/wg-training-leads @kubeflow/wg-deployment-leads @kubeflow/wg-model-registry-leads @kubeflow/kubeflow-steering-committee for the feedback.

**Comment by StefanoFioravanzo** [comment]
I agree with @kimwnasptd proposal. 

@kimwnasptd just to clarify: you are suggesting that the `kubeflow/multi-tenancy` repo would be under the Manifests WG responsibility, correct? We would probably need a "Control Plane WG" that is both responsible for `kubeflow/manifests` and `kubeflow/multi-tenancy`.

My 2 cents on:

> What do we do with components that aim to make it more streamlined to interact with K8s?

These would all make sense as standalone projects, but they cannot stay outside of one of the existing working groups just yet. I am wondering why @juliusvonkohout  suggests that we should have as few repos as possible. What is stopping us from having  `kubeflow/poddefaults` and `kubeflow/volume-management` repos? These could still be Notebooks WG subprojects, but at least have a separate lifecycle that could promote more contributions.

> TensorBoard is an ML tool, but doesn't really fit to live under kubeflow/workspaces. 

Good observation. We don't have data points as to how popular this is. I don't think that this component should stay under the Notebooks WG. I think we should:
1. Agree that indeed TF Controller cannot live under `kubeflow/workspaces`
2. Decide if we are okay with having a `kubeflow/tensorboard` repo
3. If NO (due to operational difficulties with having a separate repo, for whatever reason), then Deprecation is the only option
4. If YES, then do a call to action with a deadline. If someone is willing to maintain this repo, then we can figure out how to do so.

**Comment by juliusvonkohout** [comment]
> These would all make sense as standalone projects, but they cannot stay outside of one of the existing working groups just yet. I am wondering why @juliusvonkohout suggests that we should have as few repos as possible. What is stopping us from having `kubeflow/poddefaults` and `kubeflow/volume-management` repos? These could still be Notebooks WG subprojects, but at least have a separate lifecycle that could promote more contributions.

It adds so much overhead as Software developer, maintainer and reviewer. Just try it out yourself :-D. You will usually get lost in Processes and talking with less code and way too much communication and synchronization overhead

**Comment by thesuperzapper** [comment]
Based on some discussions today I have updated my proposed "option 3" above to suggest splitting the repo up into `kubeflow/dashboard` (could also be called `kubeflow/platform`), and `kubeflow/workspaces`.

The goal would be to build "Notebooks 2.0" in a separate branch of `kubeflow/workspaces` and eventually have it replace the need for the volumes and tensorboard controllers.

**Comment by rimolive** [comment]
> @kimwnasptd just to clarify: you are suggesting that the kubeflow/multi-tenancy repo would be under the Manifests WG responsibility, correct? We would probably need a "Control Plane WG" that is both responsible for kubeflow/manifests  and kubeflow/multi-tenancy.
This is a great idea, it can be WG Manifests responsibility as they are already working on both.

> It adds so much overhead as Software developer, maintainer and reviewer. Just try it out yourself :-D. You will usually get lost in Processes and talking with less code and way too much communication and synchronization overhead
Maybe the overhead comes from the fact that there are too much manual steps to accomplish this? If so, we can figure out a way to automate it and trigger only when we need to cut releases for Kubeflow.

**Comment by StefanoFioravanzo** [comment]
> Maybe the overhead comes from the fact that there are too much manual steps to accomplish this? If so, we can figure out a way to automate it and trigger only when we need to cut releases for Kubeflow.

Exactly! This seems something that should not block the creation of new repos, but rather encourage us to find ways to remove barriers and simplify process

**Comment by StefanoFioravanzo** [comment]
> The goal would be to build "Notebooks 2.0" in a separate branch of kubeflow/workspaces and **eventually have it replace the need for the volumes and tensorboard controllers**

@thesuperzapper what do you mean with that?

**Comment by StefanoFioravanzo** [comment]
Option 2 seems to be the most future proof and avoids confusion

**Comment by rimolive** [comment]
I'll go with Andrey's Option 2 as well. Having KF component code into `kubeflow/manifests` will not cohesively state the objective of the manifests repo.

**Comment by juliusvonkohout** [comment]
As far as i understand @andreyvelich  the second option just implies renaming kubeflow/manifests to kubeflow/platform, but still having the same content as in Option one.

I am in favor of option one with the renaming to Kubeflow/platform.

**Comment by andreyvelich** [comment]
>the second option just implies renaming kubeflow/manifests to kubeflow/platform, but still having the same content as in Option one.

From my point of view this is the right approach since `manifests` repo is the superset of all Kubeflow components to deploy `Kubeflow Platform` product. Also, these components are versioned the same as manifests.

**Comment by thesuperzapper** [comment]
I strongly believe that the manifests repo should ONLY aggregate the manifests (which are authored in the other repos).

There is no benefit to bringing code into the manifests aggregation repo. We would only create problems that make it harder to develop both the code and aggregate the manifests in a usable way.

---

The three "components" (isolatable sections of Kubeflow) that live in `kubeflow/kubeflow` right now are:

1. __Kubeflow "Dashboard":__
     - the central dashboard itself
     - profile controller
     - KFAM
     - (and probably the manifests to deploy Istio, dex, and oauth2-proxy)
 2. __Kubeflow "Workspaces":__
      - Kubeflow Notebooks (controller + UI + example images)
      - PVC Management (controller + UI)
      - Tensorboards (controller + UI)
3. __Kubeflow "Admission" (PodDefaults):__
      - This one should either be part of dashboard or have its own repo. 
      - It is used frequently in KFP, in addition to Notebooks, so should be able to be deployed separately.

Given that, I think we should create 2-3 new repos for these components, so they can be versioned on their own lifecycle:

- `kubeflow/dashboard`
- `kubeflow/workspaces`
- `kubeflow/admission` (or this can live under the dashboard repo, to reduce the number of repos)

**Comment by andreyvelich** [comment]
>I strongly believe that the manifests repo should ONLY aggregate the manifests (which are authored in the other repos).

There is no benefit to bringing code into the manifests aggregation repo. We would only create problems that make it harder to develop both the code and aggregate the manifests in a usable way.

Please can you explain what kind of problems we are going to have if we will combine platform components and manifests in a single repo ?

From my point of view the benefit of combining `manifests` and `platform` components to a single repo is to simplify process of making releases and triaging issues from our users. E.g. less GitHub repos we are going to have, less PRs and Issues will stay abandoned.
Also, `manifests` repo is only required for `Kubeflow Platform`, so combining them in a single repo make sense to me.

>Kubeflow "Workspaces":

@thesuperzapper Why do you want to PVC Management and Tensorboards to the `kubeflow/workspaces` ?

**Comment by james-jwu** [comment]
Throwing in my 2c here. I think we are trying to solve a number of issues: engineering velocity, customer perception & experience, architectural "cleanliness", and ensuring maintainability of repos. IMO not all of these issues can be solved by repo organization, but maybe we can focused on the most important issues, and find compromise on less important issues.

To the Andrey's point about "_we should create GitHub repos for Kubeflow components only when it makes sense to call the tool as an individual sub-project **[1]** and can be deployed as a standalone application.**[2]**_", I agree with the part [1]. On part [2] I think there is the chance of the "platform" growing too big, so it is not necessary to have a single "platform" repo. Rather, we could create repos for large feature areas. Today the only **user facing** feature in "platform" is central dashboard, so we could create a repo named as such. This will reduce confusion on the customers when they try to file issues, etc.

For the common components (profile controller, etc.), there are several options: leave them with central dashboard, create separate repo, move some/all to manifest repo. I think we can give the current maintainers of these components and manifest repo to decide. It may not results in the cleanest option from architectural perspective, but I want to optimize the developer/maintainer's workflow because it's not like we have a large group of developers behind each component.

Another point I want to make is whether we should separate workspace and notebook repos. IMO it may be a better product strategy to give customer confidence about product continuity from Notebook to its next version. So maybe having workspace developed in notebooks repo is a better choice from this perspective.

**Comment by thesuperzapper** [comment]
@james-jwu I 100% agree that the repos should be named by their "user-facing purpose".

That's why my initial proposal was to name the repo which contains `dashboard`/`profile-controller`/`kfam` as __`kubeflow/dashboard`__. However, to make the name more flexible (e.g. allow `admission-webhook` to be included), and open the possibility of "branding" the frontend of Kubeflow as "Kubeflow Central", I quite like the repo name __`kubeflow/central`__.

In any case, it's not possible to use the `dashboard` without `profile-controller` or `kfam` so I plan to keep them in the same repo as each other.

---

While we could develop notebooks/workspaces in the same repo, because we want to version/release them separately (and allow them to be deployed alongside each other), I think separate repos is cleaner. 

My initial thought was to use separate branches of `kubeflow/workspaces`, but that just leads to more confusion, as branches aren't very visible, and makes it harder to "archive" Notebooks 1.0 once 2.0 is mature.

---

In any case, it's clear the next steps are to:

1. Create a new __`kubeflow/workspaces`__ repo (so we can start scaffolding the Notebooks 2.0 code ASAP)
2. Create a new __`kubeflow/notebooks`__ repo (so we can migrate the Notebooks 1.0 components to it)
3. Continue to discuss the future location for dashboard/profile/kfam/poddefaults (and leave them in `kubeflow/kubeflow` until we have a decision).

If @kimwnasptd agrees on steps 1 and 2, what is the process to create those new repos @james-jwu?

**Comment by andreyvelich** [comment]
> That's why my initial proposal was to name the repo which contains dashboard/profile-controller/kfam as kubeflow/dashboard. However, to make the name more flexible (e.g. allow admission-webhook to be included), and open the possibility of "branding" the frontend of Kubeflow as "Kubeflow Central", I quite like the repo name kubeflow/central.

As @juliusvonkohout said before: https://github.com/kubeflow/kubeflow/issues/7549#issuecomment-2055717625 most of the multi-tenancy stuff is already living in `kubeflow/manifests` that is why we can move `profile-controller`, `kfam`, `admission-webhook` to the `kubeflow/manifests`. 

I think, we should gather feedback from folks in the community who is planning to maintain these components (e.g. @kubeflow/wg-manifests-leads @kubeflow/wg-training-leads).
Can we find folks in the community who can maintain these components ?


Also, what do you think about this question: **"What repo user might use if they get issues with Profile Controller ?"**

>My initial thought was to use separate branches of kubeflow/workspaces, but that just leads to more confusion, as branches aren't very visible, and makes it harder to "archive" Notebooks 1.0 once 2.0 is mature.

Why you can't use the same branches for Notebooks 1.0 and 2.0 in `kubeflow/workspaces` ?

For example, when @kimwnasptd and team worked on the new Katib UI: https://github.com/kubeflow/katib/projects/1, we created [a new directory `new-ui`](https://github.com/kubeflow/katib/tree/release-0.15/pkg/new-ui/v1beta1) and committed code there. For a while we were releasing 2 Katib UI images for old and new UI. After a while we deprecated the old Katib UI: https://github.com/kubeflow/katib/pull/2179.

**Comment by StefanoFioravanzo** [comment]
I agree with @james-jwu on keeping workspaces in the same repo as notebooks. It's much better to provide continuity by keeping both workstreams in the same repo, even if versioning/releasing may be slightly more challenging at the beginning. It makes much more sense from a product perspective and give much more clarity to users and contributors alike

**Comment by juliusvonkohout** [comment]
@andreyvelich since you asked me directly in the community call: Yes i am willing to maintain the components in  the kubeflow/platform kubeflow/workspaces split and also in Matthew Wicks kubeflow/dashboard kubeflow/workspaces
kubeflow/admission kubeflow/platform split, as well as something in between like kubeflow/platform kubeflow/workspaces kubeflow/multi-tenancy.

**Comment by thesuperzapper** [comment]
Hey @andreyvelich @james-jwu @kimwnasptd, we discussed this in the Notebooks WG meeting today, and we are happy to keep notebooks and workspaces in the same `kubeflow/notebooks` repo.

Therefore, since we are all in agreement that at least `kubeflow/notebooks` needs to exist, can we please start the process to create that repository?

**Comment by rimolive** [comment]
We need to work on a 1.9.0-rc0 release by the week of April 29th, so when working with separating notebooks from the main repo make sure this don't affect or block cut a release for notebooks for 1.0.9-rc0.

**Comment by andreyvelich** [comment]
> Hey @andreyvelich @james-jwu @kimwnasptd, we discussed this in the Notebooks WG meeting today, and we are happy to keep notebooks and workspaces in the same `kubeflow/notebooks` repo.
> 
> Therefore, since we are all in agreement that at least `kubeflow/notebooks` needs to exist, can we please start the process to create that repository?

Sure, we can start that. Let's discuss this tomorrow @kubeflow/kubeflow-steering-committee.

@kimwnasptd @thesuperzapper Do you want to use `kubeflow/notebooks` repo in the future to develop Notebooks 2.0, a.k.a Kubeflow Workspaces ?

**Comment by thesuperzapper** [comment]
Yes, that's was I was meaning by my message in https://github.com/kubeflow/kubeflow/issues/7549#issuecomment-2077768533.

Both the existing notebooks and new workspaces code will be on the same repo (and in the same branch).

**Comment by thesuperzapper** [comment]
> Sure, we can start that. Let's discuss this tomorrow @kubeflow/kubeflow-steering-committee.
> 
> @kimwnasptd @thesuperzapper Do you want to use `kubeflow/notebooks` repo in the future to develop Notebooks 2.0, a.k.a Kubeflow Workspaces ?

Yes, that's was I was meaning by my message in https://github.com/kubeflow/kubeflow/issues/7549#issuecomment-2077768533.

Both the existing notebooks and new workspaces code will be on the same repo (and in the same branch).

**Comment by andreyvelich** [comment]
We discussed this topic today during KSC call and we are happy to create this new repo `kubeflow/notebooks` to migrate Notebooks-related components.
@kimwnasptd Please can we get confirmation from you as well since you are member of WG Notebooks ?

**Comment by kimwnasptd** [comment]
@andreyvelich sounds good!

**Comment by andreyvelich** [comment]
@kubeflow/wg-notebooks-leads `kubeflow/notebooks` repo has been created: https://github.com/kubeflow/notebooks 🎉 
Thank you for doing this @zijianjoy!

**Comment by thesuperzapper** [comment]
@james-jwu @zijianjoy thanks!

Can we please also:

1. give Github write access to @thesuperzapper and @kimwnasptd on the new repo
     - (so we can approve GitHub actions and cut tags)
     - we will probably need to raise a new pr in [`kubeflow/internal-acls`](https://github.com/kubeflow/internal-acls) for this
3. ensure the branch protection rules are set up like `kubeflow/kubeflow`:
     - so that the `main` branch can not be pushed to (so people can't bypass the bot)
        - ![Screenshot 2024-03-05 at 12 19 14](https://github.com/kubeflow/kubeflow/assets/5735406/7c33a647-d790-4fbe-9434-4a69c5ed0f67)
        - so that DCO is required, to pass under "Require status checks to pass before merging" (NOTE: the option for this won't come up until we raise a PR the first time).
4. Add `kubeflow/notebooks` to the following config (so that people's PRs are not automatically self-approved):
      - https://github.com/GoogleCloudPlatform/oss-test-infra/blob/master/prow/oss/plugins.yaml#L41-L54

**Comment by thesuperzapper** [comment]
I have raised a separate PR to give @kimwnasptd and @thesuperzapper write access to the new `kubeflow/notebooks` repo:

- https://github.com/kubeflow/internal-acls/pull/677

**Comment by juliusvonkohout** [comment]
@andreyvelich what are the next steps planned then?

**Comment by andreyvelich** [comment]
@james-jwu @zijianjoy Please can you let us know if you made changes according to the @thesuperzapper comment:  https://github.com/kubeflow/kubeflow/issues/7549#issuecomment-2091871635

>@andreyvelich what are the next steps planned then?

The next steps are:

1. @kubeflow/wg-notebooks-leads should transfer Kubeflow Notebooks code to the `kubeflow/notebooks`

2. Transfer Notebooks PRs and Issues to the new repo from `kubeflow/kubeflow`.

3. We need to identify WG who can take responsibility to maintain Kubeflow Platform control-plane components:

- [access-management](https://github.com/kubeflow/kubeflow/tree/master/components/access-management) (KFAM, Auth)
- [admission-webhook](https://github.com/kubeflow/kubeflow/tree/master/components/admission-webhook) (PodDefaults)
- [centraldashboard](https://github.com/kubeflow/kubeflow/tree/master/components/centraldashboard)
- [profile-controller](https://github.com/kubeflow/kubeflow/tree/master/components/profile-controller)
- [crud-web-apps](https://github.com/kubeflow/kubeflow/tree/master/components/crud-web-apps) (UIs for: Volumes, Notebooks, Tensorboards)
- [pvcviewer-controller](https://github.com/kubeflow/kubeflow/tree/master/components/pvcviewer-controller)

Maybe we can spend a few minutes [in the tomorrow's community call](https://docs.google.com/document/d/1Wdxt1xedAj7qF_Rjmxy1R0NRdfv7UWs-r2PItewxHpE/edit) cc @jbottum

**Comment by thesuperzapper** [comment]
@andreyvelich @james-jwu @zijianjoy I have raised a PR in the `GoogleCloudPlatform/oss-test-infra` repo to require self-approval for root-level approvers.

(So we don't have driveby LGTMs accidentally mering PRs which are not ready).

https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2271

**Comment by juliusvonkohout** [comment]
For "Transfer Notebooks PRs and Issues to the new repo from kubeflow/kubeflow." maybe our GSOC Student @hansinikarunarathne can help with that @rimolive

**Comment by thesuperzapper** [comment]
@juliusvonkohout while I agree with some of these being closed as they are no longer relevant, we should actually transfer issues that are still relevant, rather than just closing them.

Some of those issues have many many likes, and its a bad experience to just close them.

Also, please leave the notebooks/dashboard/profiles issues for @kubeflow/wg-notebooks-leads to migrate once we are ready.

**Comment by juliusvonkohout** [comment]
I think I transferred  the most relevant ones for manifests/platform. For the remaining ones it probably makes sense to see how active the discussion is, especially  on closed ones. If someone responds in the issues I can investigate more.

For active relevant issues we first need a finalized migration plan from @andreyvelich to see which subcomponent goes where. e.g. kubeflow/notebooks, kubeflow/platform/manifests etc. Based on that KSC decision we can transfer the issues to the right place.

**Comment by thesuperzapper** [comment]
@andreyvelich see the top of the issue under "option 3" for the split we ended up deciding on.

We have already created `kubeflow/notebooks`, and we will make `kubeflow/dashboard` in the near future after we finish migrating the notebooks code (which we will do once there are no more planned `1.9.X` releases. 

Initially, the @kubeflow/wg-notebooks-leads will maintain the new `kubeflow/dashboard` repo, along with help from new members like @ederign and others who want to join.

**Comment by andreyvelich** [comment]
>see the top of the issue under "option 3" for the split we ended up deciding on.

We haven't decided on the splitting yet, as @james-jwu said here: https://github.com/kubeflow/kubeflow/issues/7549#issuecomment-2067090396 it is up to the component developers to decide where these components should live to simplify development and issue triaging.

>Initially, the @kubeflow/wg-notebooks-leads will maintain the new kubeflow/dashboard repo, along with help from new members like @ederign and others who want to join.

So we need to know for how long @kubeflow/wg-notebooks-leads are willing to maintain the dashboard components and what should we do with other components (profile-controller,  kfam, admission-webhook, crud-web-apps) ?

As @juliusvonkohout mentioned above: https://github.com/kubeflow/kubeflow/issues/7549#issuecomment-2072818553, he can maintain these component.

Given that these components will always have the same release schedule as Kubeflow Platform, for me it makes sense to maintain them in a single repo together with Kubeflow manifests for Distribution Owners.
E.g. we can create repo: `kubeflow/platform` for these platform-specific components. That will simplify the issue triaging and release cut. But I am open for other suggestions.

Currently, @juliusvonkohout is creating Kubeflow Platform issues in the `manifests` repo.
How will we guide users when they need to create issues in `kubeflow/manifests` or `kubeflow/dashboard` ?

**Comment by thesuperzapper** [comment]
@kimwnasptd can confirm our thinking, but as @james-jwu said in https://github.com/kubeflow/kubeflow/issues/7549#issuecomment-2067090396, the most sensible "user-facing" component in "kubeflow platform" is the central dashboard, so it makes sense to name the repo `kubeflow/dashboard` (even though it also includes profiles, kfam, and PodDefaults).

Regarding new maintainers, we always welcome new participants in the Notebooks WG(who currently maintain these components), and will progress people through the normal approver process as they show commitment to a specific component by raising PRs that get merged.

Also, as we discussed previously, we will not be combining the dashboard with the manifests, this would not make any sense and would hinder our ability to maintain both sides.

**Comment by andreyvelich** [comment]
>so it makes sense to name the repo kubeflow/dashboard (even though it also includes profiles, kfam, and PodDefaults).

How are you going to explain that users should use `kubeflow/dashboard` repo if they see issues with profile controller, etc.? Also, when users should create issue in `kubeflow/manifests` and when in `kubeflow/dashboard` ?

We need to get confirmation from [the current Notebooks WG leads](https://github.com/kubeflow/community/tree/master/wg-notebooks#organizers): @StefanoFioravanzo @kimwnasptd @thesuperzapper @elikatsis @yanniszark, that you are planning to maintain these components and triage all user-related issues for long period of time.

>this would not make any sense and would hinder our ability to maintain both sides.

Please can you explain how is it harden the component development cycle ?

**Comment by andreyvelich** [comment]
Hi Folks, do we have any updates here ? 

If @kubeflow/wg-notebooks-leads don't have enough bandwidth to drive it, should we find some volunteers from the Kubeflow community to migrate platform's components from `kubeflow/kubeflow` ?

cc @kubeflow/kubeflow-steering-committee

**Comment by thesuperzapper** [comment]
@andreyvelich current status is we are doing the "option 3" listed in the issue description: https://github.com/kubeflow/kubeflow/issues/7549#issue-2240847275

That is, we will have:

- `kubeflow/notebooks`
- `kubefow/dashboard`

However, we can't move anything until after the 1.9.0 release (and probably 1.9.1), so we are waiting.

To be ready to migrate the dashboard components once that happens, we will need @james-jwu to make us a new `kubeflow/dashboard` repo.

Note, we are already [developing the Notebook v2 in the new repo](https://github.com/kubeflow/notebooks/tree/notebooks-v2), it's in a separate `notebooks-v2` branch while we work for the first release.

**Comment by kimwnasptd** [comment]
I'm also planning to help with the migration of the code of the `notebooks-v1` components, but as @thesuperzapper mentioned let's first ensure the release is out and then we'll go full in on this.

**Comment by andreyvelich** [comment]
Great, thank you for this @kimwnasptd and @thesuperzapper.
Why we can't migrate the code before 1.9.1 release ?

Also, who is going to be listed in the OWNER file in `kubeflow/dashboard` repo ?

**Comment by terrytangyuan** [comment]
Creating `kubeflow/dashboard` repo SGTM

**Comment by kimwnasptd** [comment]
Hey folks, it's been a minute. I'd like to help with this effort now that 1.9 is sorted out and migrate the existing code between the https://github.com/kubeflow/notebooks and https://github.com/kubeflow/dashboard repos.

My first goal will be to have a PR that copies the source code of existing components and setup the CI/CD. We might need to have some discussion on registries to use for the container images (simplest one would be GH). But I'll create some tracking issues and link them here as I go.

@andreyvelich @juliusvonkohout @thesuperzapper do you think I'm missing something big before getting started?

**Comment by thesuperzapper** [comment]
@kimwnasptd I would like to get the pending 1.9.1 PRs merged before doing the migration (to avoid users having to re-raise them).

There is a list of potential PRs for 1.9.1 on the 1.9.0 tracking issue, most of them need to be rebased, and some might need changes before they can be merged:

- https://github.com/kubeflow/kubeflow/issues/7453
- [`1.9.1` Milestone](https://github.com/kubeflow/kubeflow/milestone/5)

---

Once we have done that, our GSOC person @Adembc and I did a bit of invesigation about how to best migrate the history to meet the following criteria:

1. Keep the commit history
2. Ensure GitHub correctly links the squashed `(#xxxx)` at the end of the commits to the right PR (which will now be in a different repo).

We figure out that you can mostly do this with [git-filter-repo](https://github.com/newren/git-filter-repo) and rewriting the `#xxxx` to be `kubeflow/kubeflow#xxxx` so the link still works.

**Comment by kimwnasptd** [comment]
Nice work!

Regarding the 1.9.1 milestone, let's have a plan on which of those can actually be merged and have also a deadline to not let it linger on for too long. Also with a quick look it looks like a lot of those don't fit the requirements (bug/security fixes) for a 1.9.1 but should go to 1.10

I'll reach out to Slack as well to draft a plan on the above and we can post further updates

**Comment by andreyvelich** [comment]
> Hey folks, it's been a minute. I'd like to help with this effort now that 1.9 is sorted out and migrate the existing code between the https://github.com/kubeflow/notebooks and https://github.com/kubeflow/dashboard repos.
> 
> My first goal will be to have a PR that copies the source code of existing components and setup the CI/CD. We might need to have some discussion on registries to use for the container images (simplest one would be GH). But I'll create some tracking issues and link them here as I go.
> 
> @andreyvelich @juliusvonkohout @thesuperzapper do you think I'm missing something big before getting started?

Thank you for your time @kimwnasptd! It looks great to me.
Since the `/transfer` plugin has been enabled in this PR: https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2331, can we start with issue migration at least ?

Additionally, can we move source code for components that don't need to have any changes for 1.9.1 ?

**Comment by andreyvelich** [comment]
@juliusvonkohout Can you help us with the issue transfer please ?

**Comment by thesuperzapper** [comment]
@andreyvelich @juliusvonkohout just to be clear, please don't start transfering issues until we finalize the code migration plan, as we don't want issues to end up on the wrong repo.

**Comment by andreyvelich** [comment]
> @andreyvelich @juliusvonkohout just to be clear, please don't start transfering issues until we finalize the code migration plan, as we don't want issues to end up on the wrong repo.

Why we can't start the process now since it will take time to close/transfer 195 issues @thesuperzapper ? I think, it doesn't matter for users in which repository they get response on their question/suggestion.

Do we have any timelines when are we planning to migrate code and cleanup the `kubeflow/kubeflow` repo ?

We started this issue in January 24th, and according to the @james-jwu comment here: https://github.com/kubeflow/internal-acls/pull/618#issuecomment-2043257377, we were planning to give write access for `kubeflow/kubeflow` repo for 3 months.

Any thoughts @thesuperzapper @juliusvonkohout @kimwnasptd  @kubeflow/kubeflow-steering-committee ?

**Comment by juliusvonkohout** [comment]
@thesuperzapper I will try to be careful and only migrated issues that belong to website, manifests, pipelines for now.

@kimwnasptd at the same time you also need to update the synchronization scripts in Kubeflow/manifests/hack if you move stuff to a different repository.

**Comment by suse-coder** [comment]
We use cilium gateway api. Can this also be used instead of istio?

**Comment by thesuperzapper** [comment]
I have [migrated the dashboard code to `kubeflow/dashboard`](https://github.com/kubeflow/dashboard/commit/09225c08ecafc7014afcdc550cb50e081645cc21), we will now work on getting the release process set up on the new repo before removing the code from `kubeflow/kubeflow`:

See https://github.com/kubeflow/dashboard/issues/74#issuecomment-2800157517

**Comment by thesuperzapper** [comment]
First release of `kubeflow/dashboard` has been cut!

https://github.com/kubeflow/dashboard/releases/tag/v2.0.0

**Comment by thesuperzapper** [comment]
First release of `kubeflow/notebooks` has been cut!

https://github.com/kubeflow/notebooks/releases/tag/v1.11.0
